lib/filter-stream.js

```diff
@@ -168,39 +223,52 @@ function FilterStream( blockMap, options ) {
    */
   _transform( chunk, _, next ) {
 
-    this._buffers.push( chunk )
-    this._bytes += chunk.length
+    debug( 'position', this.position )
+    debug( 'chunk', chunk.length, chunk )
 
-    if( this._bytes < this.blockSize )
-      return next()
+    this.bytesRead += chunk.length
+    this.blocksRead += chunk.length / this.blockSize
 
-    var buffer = Buffer.concat( this._buffers )
-    var offset = 0
+    if( this.currentRange == null ) {
+      debug( 'no current range' )
+      return next()
+    }
 
-    while( buffer.length - offset >= this.blockSize ) {
-      this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
-      offset = offset + this.blockSize
+    if( !this._rangeInChunk( chunk ) ) {
+      debug( 'chunk:ignore' )
```

Comment:
The emitted buffers have `position` and `address` properties added to them (see [line 175](https://github.com/resin-io-modules/blockmap/pull/27/files#diff-7c60d2150dd8ff0dfcf7b7edb6417d37R173)) – this is going to change soon to a saner way (emitting objects of the form `{ position, address, data }` to utilize hidden classes and make sure that consuming streams don't accidentally write blocks in sequence), but for backwards compatibility this PR doesn't introduce that yet.